python/tank/bootstrap/configuration_writer.py

```diff
@@ -28,6 +28,50 @@
 log = LogManager.get_logger(__name__)
 
 
+class RenameGuard(object):
```

Comment:
I'd recommend not using the verb `rename`. That's taken from `os.rename()` which is the API call, not the actual thing we're trying to do. I'd recommend using `move` instead.